src/client/client.js

```diff
@@ -245,9 +267,27 @@ export default class Client {
             onRequest();
         }
         try {
-            const response = await fetch(url, this.getOptions(options));
-            const data = await response.json();
-            return response.ok ? onSuccess(data, response) : Promise.reject(data);
+            const resp = await fetch(url, this.getOptions(options));
+            let data;
+
+            // TODO: find an 'isomorphic' way to parse response headers
+            const contentType = resp.headers.get('Content-Type') ||
+                _.first(_.get(resp.headers, 'map.content-type')) || 'unknown';
```

Comment:
Why are we looking directly at the private fields of the Headers object? Shouldn't that also still return undefined if `resp.headers.get` already returned that?